### PR TITLE
Add comprehensive form support

### LINKED ISSUE #13: Support multi checkbox group
```
<input type="checkbox" name="enabled_features[]" value="featureA" checked />Feature A<br />
<input type="checkbox" name="enabled_features[]" value="featureB" />Feature B<br />
<input type="checkbox" name="enabled_features[]" value="featureC" />Feature C<br />
```

### Expected Result

```json
{
  "enabledFeatures": ["featureA"]
}
```

or

```json
{
  "enabledFeatures": {
    "featureA": true,
    "featureB": false,
    "featureC": false
  }
}
```

### Current Result

```json
{
  "enabledFeatures[]": true
}
```

This is common at least when I was in the php world a few years ago:
https://gist.github.com/southern-hu/5424068
http://www.wastedpotential.com/html-multi-checkbox-set-the-correct-way-to-group-checkboxes/

### LINKED ISSUE #32: Remove defaults from stored options
To save space, as #31 suggests, this module could avoid saving the defaults in the storage. The setup suggested in #5 will allow users to always have access to the defaults.

### PR DESCRIPTION
**Pull request by fregante**
Fixes #13 
Fixes #32

Adds https://www.npmjs.com/package/dom-form-serializer as suggested by @AndersDJohnson in https://github.com/bfred-it/webext-options-sync/issues/13#issuecomment-495926876

This seems to work great right out of the box. The only thing I'm not totally happy about is that it brings `webext-options-sync` from 9KB to 25KB 😬

It's probably an overkill but with this I probably never have to worry about forms again since it's old and tested.

### CODE REVIEW

File: source/index.ts [hunk 1]
```diff
@@ -207,80 +198,23 @@ class OptionsSync<TOptions extends OptionsSync.Options> {
...
-			this._handleFormUpdates(target as HTMLFormElement);
+			this.set(serialize(currentTarget));
 			this._timer = undefined;
 		}, 600);
 	}
-
-	_handleFormUpdates(el: HTMLFormElement): void {
-		const {name} = el;
-		let {value} = el;
-		if (!name || !el.validity.valid) {
```
Comment (bfred-it) [inline source/index.ts]: I need to restore this check. Invalid fields should not be stored

**Comment by fregante** [comment]
The serialization could actually be replaced by:

```js
new URLSearchParams(new FormData(formElement)).toString()
// required_field_6433=&timestamp=1564044141388&issue=28&saved_reply_id=&comment%5Bbody%5D=wow%29&original-line=&path=&line=
```

However the deserialization (into an object and into the form) isn't as straightforward. The multi-checkbox setup in #13 isn't translated as 
```js
{
	'enabledFeatures': ['a', 'b']
}
```
but in
```js
{
	'enabledFeatures[]': 'a', // overridden by the second entry
	'enabledFeatures[]': 'b'
}
```

when using:

```js
const {options} = await browser.storage.sync.get();
Object.fromEntries([...new URLSearchParams(serializedData)])
```

**Comment by fregante** [comment]
I briefly tried this and it appears to work perfectly in Refined GitHub (except a bug caused by https://github.com/vadimdemedes/dom-chef/issues/38). The stored object is much smaller now that defaults aren't stored.

**Comment by fregante** [comment]
I'm merging this for now. I will update the documentation before releasing a new version. I need to specify what the options object and form should look like, mentioning `dom-form-serializer`

**Pull request merged**